src/script/components/TextInput.tsx

```diff
@@ -0,0 +1,194 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React, {useEffect} from 'react';
+import {registerReactComponent} from 'Util/ComponentUtil';
+import Icon from 'Components/Icon';
+import {CheckIcon, COLOR} from '@wireapp/react-ui-kit';
+import {CSSObject} from '@emotion/core';
+
+export interface UserInputProps {
+  disabled?: boolean;
+  errorMessage?: string;
+  isError?: boolean;
+  isSuccess?: boolean;
+  label: string;
+  name: string;
+  onCancel: () => void;
+  onChange: React.ChangeEventHandler<HTMLInputElement>;
+  onSuccessDismissed?: () => void;
+  placeholder: string;
+  value: string;
+}
+
+const SUCCESS_DISMISS_TIMEOUT = 2500;
+
+const TextInput: React.FC<UserInputProps> = ({
+  disabled,
+  errorMessage,
+  isError,
+  isSuccess,
+  label,
+  name,
+  onCancel,
+  onChange,
+  onSuccessDismissed,
+  placeholder,
+  value,
+}: UserInputProps) => {
+  const isFilled = Boolean(value);
+
+  useEffect(() => {
+    if (isSuccess && onSuccessDismissed) {
+      setTimeout(() => {
+        onSuccessDismissed();
+      }, SUCCESS_DISMISS_TIMEOUT);
+    }
+  }, [isSuccess, onSuccessDismissed]);
+
+  let changedColor = null;
+  if (isError) {
+    changedColor = 'var(--text-input-alert) !important';
+  }
+  if (isSuccess) {
+    changedColor = 'var(--green-500) !important';
+  }
+  const iconCss: CSSObject = {
+    fill: changedColor,
+    alignItems: 'center',
+    bottom: 42,
+    height: 16,
+    margin: 0,
+    padding: 0,
+    position: 'absolute',
+    right: 16,
+    width: 16,
+  };
+
+  return (
+    <div
+      css={{display: 'flex', flexDirection: 'column-reverse', paddingBottom: 26, position: 'relative', width: '100%'}}
```

Comment:
How about moving all the styles to the top of the file (or even better maybe to an external file). WDYT?

Reply:
yes, we can move it to the separate file

Reply:
done